internal/alb/ls/listener.go

```diff
@@ -250,3 +264,76 @@ func (controller *defaultController) buildDefaultActions(ctx context.Context, op
 	}
 	return buildActions(ctx, authCfg, options.IngressAnnos, backend, options.TGGroup)
 }
+
+// inferCertARNs retrieves a set of certificates from ACM that matches the ingress' hosts list
```

Comment:
it will be better to make inferCertArns an member function of controller, so it don't need to pass singtons like (acmsvc aws.ACMAPI) around.
Alternatively, i sugges make the signature below:
```func (controller *defaultController) inferCertARNs(ctx context.Context, ingress *extension.Ingress)```
So 
1.  the log object can be extracted within inferCertARNs.
1. We can pass context to ListCertificates to carry timeout info in the future.